sensorhub-core/src/main/java/org/sensorhub/impl/database/system/SystemDriverDatabase.java

```diff
@@ -82,20 +82,23 @@ protected void doStart() throws SensorHubException
         }
         
         // start auto-purge timer thread if policy is specified and enabled
-        if (config.autoPurgeConfig != null && config.autoPurgeConfig.enabled)
+        for(var autoPurgeConfig : config.autoPurgeConfig)
         {
-            var uids = Collections.unmodifiableCollection(config.autoPurgeConfig.systemUIDs);
-            final IObsSystemDbAutoPurgePolicy policy = config.autoPurgeConfig.getPolicy();
-            autoPurgeTimer = new Timer();
-            TimerTask task = new TimerTask() {
-                public void run()
-                {
-                    if (!db.isReadOnly())
-                        policy.trimStorage(db, logger, uids);
-                }
-            };
-            
-            autoPurgeTimer.schedule(task, 0, (long)(config.autoPurgeConfig.purgePeriod*1000));
+            if (autoPurgeConfig != null && autoPurgeConfig.enabled)
+            {
+                var uids = Collections.unmodifiableCollection(autoPurgeConfig.systemUIDs);
+                final IObsSystemDbAutoPurgePolicy policy = autoPurgeConfig.getPolicy();
+                autoPurgeTimer = new Timer();
```

Comment:
It would be better to create the Timer object once for all policies. (otherwise I think it will create a separate Thread for every new Timer). Then you create a new Task for each policy, and schedule them all on the same Timer.

Reply:
> It would be better to create the Timer object once for all policies. (otherwise I think it will create a separate Thread for every new Timer). Then you create a new Task for each policy, and schedule them all on the same Timer.

Thank you, I have updated this. Also, the feature has been tested with multiple policies and it does look good in the admin UI.